modules/agdc-instances/datacube-ensure-user.py

```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/env python
+#
+# Ensure that a datacube user account exists for the current user
+#
+# It can copy credentials stored in .pgpass to connect to a different database
+# Or create a new user account if required.
+#
+
+from collections import namedtuple
+import os
+import random
+import pwd
+from pathlib import Path
+import string
+import sys
+import psycopg2
+
+OLD_DB_HOST = '130.56.244.227'
+PASSWORD_LENGTH = 32
+
+DBCreds = namedtuple('DBCreds', ['host', 'port', 'database', 'username', 'password'])
+
+CANT_CONNECT_MSG = """
+Unable to connect to the database ({username}@{host}:{port}).
+"""
+
+USER_ALREADY_EXISTS_MSG = """
+An account for '{}' already exists in the Data Cube Database, but 
+we were unable to connect to it. This can happen if you have used the Data 
+Cube from raijin, and are now trying to access from VDI, or vice-versa.
+
+To fix this problem, please copy your ~/.pgpass file from the system you 
+initially used to access the Data Cube, onto the current system."""
+
+
+def print_stderr(msg):
+    print(msg, file=sys.stderr)
+
+
+def can_connect(dbcreds):
+    """ Can we connect to the database defined by these credentials? """
+    try:
+        conn = psycopg2.connect(host=dbcreds.host, port=dbcreds.port, user=dbcreds.username)
+        cur = conn.cursor()
+        cur.execute('SELECT 1;')
+        return True
+    except psycopg2.Error:
+        return False
+
+
+def find_credentials(pgpass, host):
```

Comment:
It's worth adding user as a parameter too (defaulting to current user name?), as it can have multiple entries for different roles on the same host.

(I have separate admin, backup, read-only users configured in my pgpass for the hosts)